language-plutus-core/src/Language/PlutusCore/Subst.hs

```diff
@@ -18,28 +17,28 @@ substTerm ::
   (name a -> Maybe (Term tyname name a)) ->
   Term tyname name a ->
   Term tyname name a
-substTerm tynameF nameF = hoist f
+substTerm tynameF nameF = cata f
   where
     f(VarF a bnd)         = case nameF bnd of
-      Just t  -> project t
-      Nothing -> VarF a bnd
-    f(LamAbsF a bnd ty t) = LamAbsF a bnd (substTy tynameF ty) t
-    f(TyInstF a t ty)     = TyInstF a t (substTy tynameF ty)
-    f(WrapF a bnd ty t)   = WrapF a bnd (substTy tynameF ty) t
-    f(ErrorF a ty)        = ErrorF a (substTy tynameF ty)
-    f(x)                  = x
+      Just t  -> t
+      Nothing -> Var a bnd
+    f(LamAbsF a bnd ty t) = LamAbs a bnd (substTy tynameF ty) t
+    f(TyInstF a t ty)     = TyInst a t (substTy tynameF ty)
+    f(WrapF a bnd ty t)   = Wrap a bnd (substTy tynameF ty) t
+    f(ErrorF a ty)        = Error a (substTy tynameF ty)
+    f(x)                  = embed x
```

Comment:
So, I was specifically looking for a way not to do this. IMO it obscures the intent - `cata` is a fold, but we're folding into... another version of the same fixpoint. What I *want* to do is a structure-preserving transformation where I operate on just one "unfolded" layer, and that's exactly what `hoist` gives me.

Reply:
I don't think it really does? In any case, it may be more efficient and it doesn't pollute our codebase.

Reply:
I don't see any reason to put a function called `hoist` in `PlutusPrelude` when everything it does is redundant.

Reply:
It's not equivalent to `embed cata`, it's equivalent to `cata . embed`, which is quite different!

IMO this idiom of using `embed` for the other cases is just an ad hoc version of `hoist` where you haven't defined the function.